house/house.go

```diff
@@ -0,0 +1,369 @@
+	d1, err := yaml.Marshal(append(houseDevices, iotDevices...))
+	if err != nil {
+		return err
+	}
+	err = ioutil.WriteFile(filename, d1, 0644)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func NewServer() HomeManager {
+	server := &Server{}
+	allDevices, err := readConfig("config/devices.yaml")
+	if err != nil {
+		log.Println(err)
+	}
+	for _, item := range allDevices {
+		if item.Smart {
+			iotDevices = append(iotDevices, item)
+			continue
+		}
+		houseDevices = append(houseDevices, item)
+	}
+
+	c := cron.New(cron.WithSeconds())
+	c.AddFunc("*/10 * * * * *", func() {
+		err := server.DeviceManager()
+		if err != nil {
+			log.Println(err)
+		}
+	})
+
+	c.AddFunc("*/10 * * * * *", func() {
+		err := server.iotStatusManager()
+		if err != nil {
+			log.Println(err)
+		}
+	})
+
+	metrics = make(map[string]prometheus.Gauge)
+	for _, item := range append(iotDevices, houseDevices...) {
+		log.Println(fmt.Sprintf("Registering metric for %s", item.Name))
+		server.registerMetric(*item)
+	}
+
+	c.Start()
+	server.recordMetrics()
+	return server
+}
+
+func (s *Server) registerMetric(item device) {
+	if metrics[item.Name] == nil {
+		metrics[item.Name] = promauto.NewGauge(prometheus.GaugeOpts{
+			Name: "home_detector_device",
+			Help: "Device in home",
+			ConstLabels: prometheus.Labels{
+				"name": strings.ReplaceAll(item.Name, " ", "_"),
+				"mac":  item.Id.Mac,
+				"ip":   item.Id.Ip,
+			},
+		})
+		metrics[item.Name+"_lastseen"] = promauto.NewGauge(prometheus.GaugeOpts{
+			Name: "home_detector_device_lastseen",
+			Help: "Device in home last seen",
+			ConstLabels: prometheus.Labels{
+				"name": strings.ReplaceAll(item.Name, " ", "_"),
+				"mac":  item.Id.Mac,
+				"ip":   item.Id.Ip,
+			},
+		})
+	}
+}
+
+func (s *Server) recordMetrics() {
+	go func() {
+		for {
+			homeCounter := 0
+			for _, person := range houseDevices {
+				if !person.Away && person.Person {
+					homeCounter++
+				}
+			}
+			people_home.Set(float64(homeCounter))
+
+			for _, item := range append(iotDevices, houseDevices...) {
+				state := 1
+				if item.Away {
+					state = 0
+				}
+				if metrics[item.Name] != nil {
+					metrics[item.Name].Set(float64(state))
+				}
+				if metrics[item.Name+"_lastseen"] != nil {
+					metrics[item.Name+"_lastseen"].Set(float64(item.LastSeen))
+				}
+			}
+			time.Sleep(2 * time.Second)
+		}
+	}()
+}
+func (s *Server) adjustLights(lightGroup string, brightness string) error {
+	_, err := s.callAssistant("set " + lightGroup + " lights to " + brightness + "% brightness")
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (s *Server) callAssistant(command string) (*string, error) {
+	if *debug {
+		return &command, nil
+	}
+	return assistant.Call(command)
+}
+
+func (s *Server) DeviceDetectState(item device) int64 {
+	lastSeen := int64(time.Now().Unix()) - item.LastSeen
+	return lastSeen
+}
+
+func (s *Server) Address(ctx context.Context, in *pb.AddressRequest) (*pb.Reply, error) {
```

Comment:
warning: cyclomatic complexity 25 of function (*Server).Address() is high (> 15) (gocyclo)